genesyscloud/group/resource_genesyscloud_group.go

```diff
@@ -0,0 +1,333 @@
+
+	util.RetryWhen(util.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Directory occasionally returns version errors on deletes if an object was updated at the same time.
+		log.Printf("Deleting group %s", name)
+		resp, err := gp.deleteGroup(ctx, d.Id())
+		if err != nil {
+			return resp, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to delete group %s: %s", name, err), resp)
+		}
+		return nil, nil
+	})
+
+	return util.WithRetries(ctx, 60*time.Second, func() *retry.RetryError {
+		group, resp, err := gp.getGroupById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(resp) {
+				log.Printf("Group %s deleted", name)
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("Error deleting group %s: %s", d.Id(), err))
+		}
+
+		if group.State != nil && *group.State == "deleted" {
+			log.Printf("Group %s deleted", name)
+			return nil
+		}
+
+		/*
+		  This extra delete call is being added here because of  DEVTOOLING-485.  Basically we are in a transition
+		  state with the groups API.  We have two services BEVY and Directory that are managing groups.  Bevy is dual
+		  writing to directory.  However, Directory always returns a 200 on the delete and then fails asynchronously.
+		  As a result the delete sometimes does not occur and then we just keep picking it up as it has.
+		  After talking with Joe Fruland, the team lead for directory we are putting this extra DELETE in here
+		  to keep trying to delete in case of this situation.
+		*/
+		resp, err = gp.deleteGroup(ctx, d.Id())
+		if err != nil {
+			log.Printf("Error while trying to delete group %s inside of the delete retry.  Correlation id of failed call %s",
+				name, resp.CorrelationID)
+		}
+
+		return retry.RetryableError(fmt.Errorf("Group %s still exists", d.Id()))
+	})
+}
+
+func updateGroupMembers(ctx context.Context, d *schema.ResourceData, sdkConfig *platformclientv2.Configuration) diag.Diagnostics {
+	gp := getGroupProxy(sdkConfig)
+	if d.HasChange("member_ids") {
+		if membersConfig := d.Get("member_ids"); membersConfig != nil {
+			configMemberIds := *lists.SetToStringList(membersConfig.(*schema.Set))
+			existingMemberIds, err := getGroupMemberIds(ctx, d, sdkConfig)
+			if err != nil {
+				return err
+			}
+
+			maxMembersPerRequest := 50
+			membersToRemoveList := lists.SliceDifference(existingMemberIds, configMemberIds)
+			chunkedMemberIdsDelete := chunks.ChunkBy(membersToRemoveList, maxMembersPerRequest)
+
+			chunkProcessor := func(membersToRemove []string) diag.Diagnostics {
+				if len(membersToRemove) > 0 {
+					if diagErr := util.RetryWhen(util.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+						_, resp, err := gp.deleteGroupMembers(ctx, d.Id(), strings.Join(membersToRemove, ","))
+						if err != nil {
+							return resp, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to remove members from group %s: %s", d.Id(), err), resp)
```

Comment:
As part of this work I went through and started using the diagnostics pattern to get the log format in the way we want.